### PR TITLE
Alert component with added link is created

### PR DESCRIPTION
**Pull request by kterekh**
Alert component is created: https://trello.com/c/i6rgDHzF/9-create-alert-component

@silchencko please review.

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -1,31 +1,41 @@
 <template>
   <div id="app">
-    <Button title="button" v-bind:color="btnColor.Primary" size="lg" v-bind:is-disabled="true"
-            v-bind:action=" () => showAlert('Hello!')"></Button>
-    <Button title="button" v-bind:color="btnColor.Secondary" size="lg" v-bind:is-disabled="false"></Button>
+    <app-button title="button" v-bind:color="btnColor.Primary" size="lg" v-bind:is-disabled="true"
```
Comment (silchencko) [inline src/App.vue:3]:
```suggestion
    <app-button
      title="button"
      v-bind:color="btnColor.Primary"
      size="lg" v-bind:is-disabled="true"
      v-bind:action=" () => showAlert('Hello!')" />
```
My suggestion is to align attributes. If you have nothing between `<app-button...>` and `</app-button>`, you can shorten it: `<app-button... />`

File: src/common/Alert.vue [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  },
+  computed: {
+    alClass: function (): string {
+      return `alert alert-${this.color}`
+    }
+  }
+}
+</script>
+
+<style lang="scss">
```
Comment (silchencko) [inline src/common/Alert.vue:45]: remove `<style lang="scss">.  </style>` if you don't use it

File: src/common/Alert.vue [hunk 3]
```diff
@@ -0,0 +1,47 @@
+<template>
+  <div>
+    <div class="alert" v-bind:class="alClass">
+      {{title}}
+      <a v-bind:href="linkValue">{{linkTitle}}</a>
```
Comment (silchencko) [inline src/common/Alert.vue:5]: I'd add `v-if="linkValue"` to render it only when `linkValue` and `linkTitle` are defined

File: src/App.vue [hunk 4]
```diff
@@ -1,31 +1,41 @@
...
+    'app-alert': Alert
   },
   data: function () {
     return {
-      btnColor: btnColor
+      btnColor: btnColor,
+      alertColor: alertColor
     }
   },
+  link: true,
```
Comment (silchencko) [inline src/App.vue:31]: Do you use `link: true ` ?

File: src/common/Alert.vue [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    },
+    role: {
+      type: String,
+      required: true
+    },
+    title: {
+      type: String,
+      required: true
+    },
+    linkValue: {
```
Comment (silchencko) [inline src/common/Alert.vue:30]: Maybe it makes sense to keep link value and link title in one object:
```
link: {
  title: 'link',
  value: 'http://link.com'
}
```
If you use it, you will need an interface with title and value properties
This is up to you

File: src/common/Alert.vue [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+      name: 'Alert'
+    }
+  },
+  props: {
+    color: {
+      type: String,
+      required: true
+    },
+    role: {
+      type: String,
```
Comment (silchencko) [inline src/common/Alert.vue:23]: hm, do you use `role` somewhere in Alert component?

**Review by silchencko — COMMENTED** [review]
@kterekh Looks great! Good job! And good progress with the second component!
I left some comments to make it even better.
Please don't hesitate to use `npm run lint --fix` before commit

File: src/common/Alert.vue (hunk 6 above)
Comment (kterekh) [reply src/common/Alert.vue:23]: No, thanks for paying attention to details.  It was as part of my first idea to implement this alert. Deleted it.

File: src/App.vue (hunk 1 above)
Comment (kterekh) [reply src/App.vue:3]: Make sense, thanks!

File: src/common/Alert.vue (hunk 5 above)
Comment (kterekh) [reply src/common/Alert.vue:30]: Let's discuss it on our call today.

File: src/App.vue (hunk 4 above)
Comment (kterekh) [reply src/App.vue:31]: Let's discuss it on our call today.

File: src/common/Alert.vue (hunk 3 above)
Comment (kterekh) [reply src/common/Alert.vue:5]: Done.

File: src/common/Button.vue [hunk 7]
```diff
@@ -8,39 +8,20 @@
...
-  computed: {
-    btnClass: function (): string {
-      return `btn-${this.color} btn-${this.size}`
-    }
+@Component
+export default class Button extends Vue {
+  @Prop() color!: string
+  @Prop() size!: string
+  @Prop() title!: string
+  @Prop() isDisabled!: string
```
Comment (silchencko) [inline src/common/Button.vue:18]: isDisabled - boolean

**Review by silchencko — CHANGES_REQUESTED** [review]

File: src/common/Button.vue [hunk 8]
```diff
@@ -8,39 +8,20 @@
...
+  @Prop() action!: string
+
+  name = 'Button'
```
Comment (silchencko) [inline src/common/Button.vue:21]: You don't need name

**Review by silchencko — APPROVED** [review]

File: src/App.vue [hunk 9]
```diff
@@ -1,31 +1,46 @@
...
-    showAlert: function (message: string) {
-      alert(message)
-    }
+@Component({
+  components: { 'app-button': Button, 'app-alert': Alert }
+})
+export default class App extends Vue {
+  btnColor = btnColor
+  alertColor = alertColor
+  linkAvailable = true
```
Comment (silchencko) [inline src/App.vue:38]: : boolean

File: src/common/Alert.vue [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { Component, Prop, Vue } from 'vue-property-decorator'
+
+@Component
+export default class Alert extends Vue {
+  @Prop() color!: string
```
Comment (silchencko) [inline src/common/Alert.vue:15]: @Prop({ required: true })
color!: alertColor

**Pull request merged**